src/pages/auth/AsyncButton.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import {Text, ActivityIndicator} from 'react-native';
+import styles from './AsyncButton.scss';
+import {Button} from "react-native-paper";
+
+interface AsyncButtonProps {
+    loading: boolean;
+    onPress: () => void;
+    buttonText: string;
+}
+
+const AsyncButton: React.FC<AsyncButtonProps> = ({loading, onPress, buttonText}) => {
```

Comment:
how it looks like in practice?

Reply:
If the data is being loaded the button display the loading icon instead of the text:

https://github.com/user-attachments/assets/1e7a875e-8016-4104-813f-fcaef0cd9b93